prov/tcp/src/tcpx_ep.c

```diff
@@ -68,22 +68,19 @@ static ssize_t tcpx_recvmsg(struct fid_ep *ep, const struct fi_msg *msg,
 	if (!posted_rx)
 		return -FI_ENOMEM;
 
-	posted_rx->ep = tcpx_ep;
 	posted_rx->flags = flags;
 	posted_rx->context = msg->context;
-	posted_rx->iov_cnt = msg->iov_count;
-	len = 0;
+	posted_rx->msg_data.iov_cnt = msg->iov_count;
 
 	for (i = 0 ; i < msg->iov_count; i++) {
-		iov.iov.addr = (uintptr_t) msg->msg_iov[i].iov_base;
-		iov.iov.len = msg->msg_iov[i].iov_len;
-		len += msg->msg_iov[i].iov_len;
-		memcpy(&posted_rx->iov[i], &iov, sizeof(iov));
+		iov.addr = (uintptr_t) msg->msg_iov[i].iov_base;
+		iov.len = msg->msg_iov[i].iov_len;
+		memcpy(&posted_rx->msg_data.iov[i], &iov, sizeof(iov));
```

Comment:
We should set msg_dat.iov[i] directly, rather than setting iov and then doing a memcpy.

Reply:
Thanks. Done.